--- a/windows/audioplayers_windows_plugin.cpp
+++ b/windows/audioplayers_windows_plugin.cpp
@@ -50,7 +50,7 @@
   static inline BinaryMessenger* binaryMessenger;
   static inline std::unique_ptr<MethodChannel<EncodableValue>> methods{};
   static inline std::unique_ptr<MethodChannel<EncodableValue>> globalMethods{};
-  static inline std::unique_ptr<EventStreamHandler<>> globalEvents{};
+  static inline EventStreamHandler<>* globalEvents = nullptr;
 
   // Called when a method is called on this plugin's channel from Dart.
   void HandleMethodCall(const MethodCall<EncodableValue>& method_call,
@@ -92,11 +92,9 @@
       [plugin_pointer = plugin.get()](const auto& call, auto result) {
         plugin_pointer->HandleGlobalMethodCall(call, std::move(result));
       });
-  globalEvents = std::make_unique<EventStreamHandler<>>();
-  auto _obj_stm_handle =
-      static_cast<StreamHandler<EncodableValue>*>(globalEvents.get());
-  std::unique_ptr<StreamHandler<EncodableValue>> _ptr{_obj_stm_handle};
-  _globalEventChannel->SetStreamHandler(std::move(_ptr));
+  auto globalEventHandler = std::make_unique<EventStreamHandler<>>();
+  globalEvents = globalEventHandler.get();
+  _globalEventChannel->SetStreamHandler(std::move(globalEventHandler));
 
   registrar->AddPlugin(std::move(plugin));
 }
